### PR TITLE
Mutationobserver, closes #397

### LINKED ISSUE #397: MutationObserver polyfill
Hello!
There are some tasks related with handling element mutations, able to be polyfilled in IE8+.
There’s a polyfill-like lib, [mutation-observer](https://github.com/webmodules/mutation-observer), which does this, but there’s no such polyfill in the service.
Is it reasonable to add `MutationObserver` polyfill?
Thank you.

### PR DESCRIPTION
**Pull request by triblondon**

### CODE REVIEW

File: polyfills/MutationObserver/config.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		"android": "<4.4",
+		"firefox_mob": "<14",
+		"ie_mob": "<11",
+		"op_mob": "<15",
+		"ios_saf": "<7"
+	},
+	"dependencies": [
+		"WeakMap"
+	],
+	"repo": "https://github.com/webcomponents/webcomponentsjs",
```
Comment (JakeChampion) [inline polyfills/MutationObserver/config.json:18]: Is this the correct repository?

Comment (JakeChampion) [inline polyfills/MutationObserver/config.json:18]: Seems so. 👍

**Comment by drKnoxy** [comment]
💭 i swear i hit pollyfill.io?feature=MutationObserve in ie10 a week ago and it returned  https://github.com/webmodules/mutation-observer 

would love to get this in...

**Comment by thedrew12** [comment]
👍

**Comment by triblondon** [comment]
Some test failures here need investigation

**Comment by drKnoxy** [comment]
Some investigation from the failed test
Two Failed tests for ie 9

```
// subtree testing near L616
"message":"expected [] to sort of equal [ {} ]",
"stack":null,
"name":"MutationObserver MutationObserver childList subtree",
"result":false

// near L656
"message":"expected 0 to equal 2",
"stack":null,
"name":"MutationObserver MutationObserver childList both direct and subtree",
"result":false
```

"uaString":"Mozilla/5.0 (compatible; MSIE 9.0; Windows NT 6.1; Trident/5.0; SLCC2; .NET CLR 2.0.50727; .NET4.0C; .NET4.0E)",

**Comment by JakeChampion** [comment]
@triblondon Is this needed by FT?

**Comment by triblondon** [comment]
Not that I'm aware, all the browsers we support for JS have MutationObserver.  I wanted it because it was a useful part of the implementation of IntersectionObserver but Philip ended up doing that in a different way.  This PR is almost there but has some IE test failures.  Some community members have declared an interest - @drKnoxy and @thedrew12 would you like to pick this up?  Otherwise I guess we can close it.

**Comment by triblondon** [comment]
Over in #397, @hipkiss97 said they wanted to look into this issue, so I'll hold it open for another couple of weeks.

**Comment by sheriff91** [comment]
The test shows that it failed on ie11. Why? MuOb's are available in ie11. Thoughts?

**Comment by JakeChampion** [comment]
@hipkiss91 It seems we are [targeting ie 11 with this polyfill](https://github.com/Financial-Times/polyfill-service/pull/695/files#diff-362e33cd07e824502bb9d5f615069c14R5). Should we not be doing this?

**Comment by triblondon** [comment]
IIRC, the problem is that MOs are not spec-compliant in IE11, so the polyfill is there to correct for the buggy behaviour.

**Comment by hipkiss91** [comment]
Apologies for some questions that I may have until I figure this out. I haven't worked with test suites before, so I'm wondering what js test lib does the polyfill project use?

**Comment by sheriff91** [comment]
Is it just mocha being used or is it also using proclaim or qunit or expected.js?

**Comment by JakeChampion** [comment]
@hipkiss91 mocha provides the `decribe` and `it` functions. `proclaim` provides the assertions. They are all we are using :-)

**Comment by sheriff91** [comment]
Is exepcted.js for expect() methods? Apologies if it's a dumb question, not used testing frameworks before.  
But if the answer is true then for the failures in [IE11](https://saucelabs.com/tests/d10a17a6291f49be8edabcc093e07e05) just need to add `expect = require('expect.js');` to the top of the tests file.  

Also there's no .env file? So I can't configure the `SAUCE_USER_NAME` and `SAUCE_API_KEY` to allow me to test?

**Comment by JakeChampion** [comment]
@hipkiss91 It seems these tests are rather old and use the expectation library we used to use. We have since switched to [`proclaim`](https://github.com/rowanmanning/proclaim), most of the api is the same as our old assertion library, you will need to change the`expect`s to `proclaim`s.

**Comment by hipkiss91** [comment]
Some I've made some changes to the tests.js file - replacing the expects/asserts with proclaims. But I;m having an issue with expectRecord(). I'm not sure what I should be changing this to. What was the old lib used?

**Comment by sheriff91** [comment]
@triblondon @JakeChampion How do I submit my updates to circleCI for testing? Or is that something you guys do?

**Comment by JakeChampion** [comment]
@hipkiss91 That's something only we can do, will do it now :-)

**Comment by JakeChampion** [comment]
@hipkiss91 Can you make a PR with your changes?

**Comment by sheriff91** [comment]
@JakeChampion Done: https://github.com/Financial-Times/polyfill-service/pull/937
:)

**Comment by JakeChampion** [comment]
@hipkiss91 I fixed the ESLint errors and sorted out the `proclaim` issues in the tests.

**Comment by JakeChampion** [comment]
@hipkiss91 This PR now has all the tests running correctly, I can see that the polyfill/tests fails for IE9. If you or another contributor can get the tests passing in IE 9 then I think we would be able to merge this into master :-)

**Comment by sheriff91** [comment]
@JakeChampion That's awesome news. Apologies if I seemed a little dumb with this. First time I've contributed to something, so it's nice you all bared with me! Really not sure what happened with my indentation. I have it set in sublime text correctly - weird. Thanks for fixing it. I can take another look, see what I can come up with. Again, thanks :)

**Comment by triblondon** [comment]
Hmm, I don't know why my earlier commit didn't have my author metadata

**Comment by triblondon** [comment]
@JakeChampion shall we close this one?

**Comment by JakeChampion** [comment]
This feature was added in #1056

**Pull request closed without merging**